run-in-retro-term.sh

```diff
@@ -0,0 +1,169 @@
+    "    java" \
+    "    javascript" \
+    "    perl" \
+    "    python" \
+    "    ruby" \
+    "" \
+    "    bash [-l language]  select human language of tetris (only work with bash):"
+}
+
+usage_message_2() {
+printf "%s\n" \
+    "" \
+    "english" \
+    "$available_languages" \
+    ""
+}
+
+usage_message_3() {
+printf "%s\n" \
+    "" \
+    " -c    console profile" \
+    "    1      Default Amber" \
+    "    2      Monochrome Green" \
+    "    3      Green Scanlines" \
+    "    4      Default Pixelated" \
+    "    5      Apple ][" \
+    "    6      Vintage" \
+    "    7      IBM Dos" \
+    "    8      IBM 3278" \
+    "    9      Futuristic"
+}
+
+# usage
+usage() {
+    usage_message_0
+    usage_message_1
+    usage_message_2
+    usage_message_3
+    exit 0
+}
+
+# check cool-retro-term and xdg-open commands
+check_commands() {
+    local message=()
+    command -v xdg-open &>$stdout || message+="xdg-open could not be found, please install it"
+    command -v cool-retro-term &>$stdout || message+="cool-retro-term could not be found, please install it"
+    [ ${#message} == 0 ] && return 0
+    printf "%s\n" "${message[@]}"
+    echo "Are you sure you are using GNU operating system ;-)?"
+    exit 1
+}
+
+# check option arguments
+check_hl() {
+    [ "$pl" != "bash" ] && {                                                            # default language is english, see 10th line
+        echo "only bash support internationalization, aborting."
+        usage_message_2
+        exit 1
+    }
+}
+
+check_pl() {
+    [[ "$pl" =~ ^(bash|c|java|javascript|perl|python|ruby)$ ]] || {                     # check pl
+        echo "$pl is not supported yet, aborting."
+        usage_message_1
+        usage_message_3
+        exit 1                                                                          # stop if wrong pl given
+    }
+}
+
+check_console_profile() {
+    [[ $console_profile =~ ^(1|2|3|4|5|6|7|8|9)$ ]] || {                                # check console profile
+        echo "console profile $console_profile not finded."
+        usage_message_3
+        exit 1
+    }
+}
+
+# music loop
+run_music() {
+    while : ; do                                                                        # play music with xdg-open
+        xdg-open "$script_dir/media/tetris-theme.ogg" &>$stdout
+        sleep 85                                                                        # sleep 85, 85 is total of music time.
+    done
+}
+
+# functions for execute tetris.(*)                                                      # set exec_tetris for user specification
+load_exec() {
+    [ "$pl" == "bash" ] && exec_tetris="$script_dir/tetris.sh -l $hl"
+    [ "$pl" == "c" ] && exec_tetris="gcc $script_dir/tetris.c -o /tmp/tetris && /tmp/tetris"
+    [ "$pl" == "java" ] && exec_tetris="javac -d /tmp/ $script_dir/Tetris.java && java -classpath /tmp Tetris"
+    [ "$pl" == "javascript" ] && exec_tetris="node $script_dir/tetris.js"
```

Comment:
For all interpreted languages we don't need to explicitly mention interpreter, let's use just
```
$script_dir/tetris.ext
```
similar to bash.

Reply:
Please also remove node, tetris.js has proper shebang line.

Reply:
Also let's put those pairs in the associative array.
```
declare -A lang_to_exec=(
    [bash]="$script_dir/tetris.sh -l $hl"
    [c]="gcc $script_dir/tetris.c -o /tmp/tetris && /tmp/tetris"
    ...
)
```